qa/rpc-tests/wallet_deprecation.py

```diff
@@ -76,31 +74,39 @@ def verify_disabled(self, function):
                 "failed with '%s'" % errorString if len(errorString) > 0 else "succeeded",
             ))
 
-    def test_case(self, start_mode, features_to_allow, expected_state):
+    def test_case(self, start_mode, features_to_allow, expected_state, default_enabled, default_disabled):
         stop_nodes(self.nodes)
         wait_bitcoinds()
         start_mode(features_to_allow)
 
-        for function in DEFAULT_ENABLED:
-            expected_state(function)
-        for function in DEFAULT_DISABLED:
-            expected_state(function)
+        for function in default_enabled:
+            if function in TESTABLE_FEATURES:
+                expected_state(function)
+        for function in default_disabled:
+            if function in TESTABLE_FEATURES:
+                expected_state(function)
 
     def run_test(self):
+        dep_info = self.nodes[0].getdeprecationinfo()
+        default_enabled = dep_info['deprecated_features']
+        default_disabled = dep_info['disabled_features']
+
         # RPC methods that are deprecated but enabled by default should succeed
-        for function in DEFAULT_ENABLED:
-            self.verify_enabled(function)
+        for function in default_enabled:
+            if function in TESTABLE_FEATURES:
+                self.verify_enabled(function)
 
         # RPC methods that are deprecated and not enabled by default should fail
-        for function in DEFAULT_DISABLED:
-            self.verify_disabled(function)
+        for function in default_disabled:
+            if function in TESTABLE_FEATURES:
+                self.verify_disabled(function)
 
         for start_mode in (self.setup_network_with_args, self.setup_network_with_config):
             # restart with a specific selection of deprecated methods enabled
-            self.test_case(start_mode, DEFAULT_DISABLED, self.verify_enabled)
+            self.test_case(start_mode, default_disabled, self.verify_enabled, default_enabled, default_disabled)
```

Comment:
Hmm, this slightly changes the semantics of the test. Previously we were only enabling a subset of the default-disabled features, whereas now we always enable all of them. But then, we were only checking the ones we were testing, so I guess the difference in semantics wasn't necessarily part of the tested behaviour (although interactions between the subset being re-enabled and the full set might have been caught).